File: patch_queries.js
const fs = require('fs');

const dashboardPath = 'src/app/(authenticated)/dashboard/page.tsx';
let dashboard = fs.readFileSync(dashboardPath, 'utf8');

if (!dashboard.includes('gte(')) {
  const target = `    .neq('status', 'draft')
    .order('created_at', { ascending: false });`;
  
  const replace = `    .neq('status', 'draft')
    .gte('created_at', new Date(Date.now() - 90 * 24 * 60 * 60 * 1000).toISOString())
    .order('created_at', { ascending: false });`;
    
  dashboard = dashboard.replace(target, replace);
  fs.writeFileSync(dashboardPath, dashboard);
  console.log('Dashboard query optimized.');
}

const contentsPath = 'src/app/(authenticated)/contents/page.tsx';
if (fs.existsSync(contentsPath)) {
  let contents = fs.readFileSync(contentsPath, 'utf8');
  if (!contents.includes('gte(')) {
    const target = `    .neq('status', 'draft')
    .order('created_at', { ascending: false });`;
    
    const replace = `    .neq('status', 'draft')
    .gte('created_at', new Date(Date.now() - 180 * 24 * 60 * 60 * 1000).toISOString())
    .order('created_at', { ascending: false });`;
      
    contents = contents.replace(target, replace);
    fs.writeFileSync(contentsPath, contents);
    console.log('Contents query optimized.');
  }
}

File: patch_queries.py
import re

def patch_file(filepath, days):
    with open(filepath, 'r', encoding='utf-8') as f:
        code = f.read()

    # Find the query
    target = r"(\.neq\('status',\s*'draft'\))\s*\n\s*(\.order\('created_at')"
    
    replace = r"\1\n    .gte('created_at', new Date(Date.now() - " + str(days) + r" * 24 * 60 * 60 * 1000).toISOString())\n    \2"
    
    if ".gte('created_at'" not in code:
        code = re.sub(target, replace, code)
        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(code)
        print(f"Patched {filepath}")
    else:
        print(f"Already patched {filepath}")

patch_file('src/app/(authenticated)/dashboard/page.tsx', 90)
patch_file('src/app/(authenticated)/contents/page.tsx', 180)
